src/workspaces/useWorkspacesData.ts

```diff
@@ -0,0 +1,35 @@
+import { useAutoLoadedData } from 'src/libs/ajax/loaded-data/useAutoLoadedData';
+import { UseLoadedDataArgs } from 'src/libs/ajax/loaded-data/useLoadedData';
+import { FieldsArg } from 'src/libs/ajax/workspaces/providers/WorkspaceProvider';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+
+export interface UseWorkspacesStateResult {
+  workspaces: WorkspaceWrapper[];
+  refresh: () => Promise<void>;
+  loading: boolean;
+}
+
+export type UseWorkspacesDataArgs = UseLoadedDataArgs<WorkspaceWrapper[]> & {
+  getData: () => Promise<WorkspaceWrapper[]>;
+};
+
+export type UseWorkspacesState = (fields?: FieldsArg, stringAttributeMaxLength?: number) => UseWorkspacesStateResult;
+
+/**
+ * handles conversion of useAutoLoadedData mechanics to existing hook return expectations
+ * @param args - object {} must include getData method, and optionally onSuccess and/or onError
+ */
+export const useWorkspacesData = (args: UseWorkspacesDataArgs): UseWorkspacesStateResult => {
```

Comment:
It's unclear what the difference between `useWorkspaces` and `useWorkspaceData` is. I think I'd rather get rid of this layer altogether to avoid that confusion.

Reply:
if I do the follow-up refactor to just have useWorkspaces return LoadedResult<T>  (or that minus the "None" state as you suggest), then I am pretty sure useWorkspacesData can go away then.

Reply:
useWorkspacesData removed from solution.